bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/console/IdfProcessConsole.java

```diff
@@ -0,0 +1,189 @@
+/*******************************************************************************
+ * Copyright 2023 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.debug.gdbjtag.openocd.dsf.console;
+
+import java.nio.charset.Charset;
+
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.preferences.IDebugPreferenceConstants;
+import org.eclipse.debug.ui.IDebugUIConstants;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.resource.JFaceResources;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
+import org.eclipse.ui.console.IOConsole;
+import org.eclipse.ui.console.IOConsoleInputStream;
+import org.eclipse.ui.console.IOConsoleOutputStream;
+
+/**
+ * Idf process console created for customization of the process output to filter and color code
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+@SuppressWarnings("restriction")
+public class IdfProcessConsole extends IOConsole implements IPropertyChangeListener
```

Comment:
Could you please mention if there is any original class for this and what part of the code is modified in that so that it will be easy for us from the code maintenance point of view